kt/godot-library/src/main/kotlin/godot/global/GD.kt

```diff
@@ -0,0 +1,68 @@
+package godot.global
+
+import godot.Object
+import godot.RandomNumberGenerator
+import godot.Resource
+import godot.ResourceLoader
+import godot.core.Dictionary
+import godot.core.NodePath
+import godot.core.PoolByteArray
+import godot.core.PoolColorArray
+import godot.core.PoolIntArray
+import godot.core.PoolRealArray
+import godot.core.PoolStringArray
+import godot.core.PoolVector2Array
+import godot.core.PoolVector3Array
+import godot.core.VariantArray
+import godot.util.nullptr
+
+
+object GD : GDMath, GDCore, GDRandom, GDPrint {
+    override val rng = RandomNumberGenerator()
+
+    /** Asserts that the condition is true.
+    If the condition is false, an error is generated and the program is halted until you resume it.
+    Only executes in debug builds. Use it for debugging purposes, to make sure a statement is true during development. */
+    fun assert(condition: Boolean, message: String = "") = assert(condition) { message }
+
+    /** Returns whether instance is a valid object (e.g. has not been deleted from memory).*/
+    fun isInstanceValid(instance: Object): Boolean {
+        return instance.rawPtr != nullptr
```

Comment:
Here I'm unsure if this really is enough. If the pointer on the cpp side is gone this value is not updated as far as i can see.
Maybe some of you cpp devs have an idea how we can properly implement that?

Reply:
For this one, you should probably use this function from the GarbageCollector:
https://github.com/utopia-rise/godot-jvm/blob/bf09d77505502fedacdd2ec0b3290d2adeb23509/kt/godot-runtime/src/main/kotlin/godot/core/GarbageCollector.kt#L188

Just checking if the ptr is null is not enough. Freeing the memory doesn't make the pointers to its location null.

Reply:
Good catch

Reply:
Added the suggested check. The `rawPointer` we store is the right one to use for the `checkInstance` check i assume?